activerecord/lib/active_record/querying.rb

```diff
@@ -11,7 +11,8 @@ module Querying
     delegate :find_each, :find_in_batches, :in_batches, to: :all
     delegate :select, :group, :order, :except, :reorder, :limit, :offset, :joins, :left_joins, :left_outer_joins, :or,
              :where, :rewhere, :preload, :eager_load, :includes, :from, :lock, :readonly, :extending,
-             :having, :create_with, :distinct, :references, :none, :unscope, :merge, to: :all
+             :having, :create_with, :distinct, :references, :none, :unscope, :merge,
+             :reselect, :reselect!, to: :all
```

Comment:
`reselect` and `reselect!` do not have any difference. I think we should rename `reselect!` to `_reselect!` and only access it internally for `reselect`. Thoughts?

Reply:
My bad. `reselect!` and `reselect` have a difference. `reselect!` operates on relation in-place instead of copying.

Reply:
reselect! is not needed to delegate since that is not public API.

Reply:
@kamipo : Will it be okay if I rename `reselect!` to `_reselect!`? 

`select` has similar implementation and we can maintain consistency in that case.

Reply:
IMO it is fine as is.